code/lib/types/src/modules/core-common.ts

```diff
@@ -317,7 +317,7 @@ export interface StorybookConfig {
    *
    * @example `['./src/*.stories.@(j|t)sx?']`
    */
-  stories: StoriesEntry[];
+  stories: StoriesEntry[] | ((list: StoriesEntry[]) => Promise<StoriesEntry[]>);
```

Comment:
FYI: This function-type API is more for internal use, and not meant to be used in `main.ts` files.
I'd love to understand why do you want this?

Technically this should likely be: `PresetProperty<StoriesEntry[]>`.

We've been talking about this in the team, and we would actually maybe like to split the `StorybookConfig` type into a type that's meant for public use, and a private one.

The private interface would have all properties be `PresetProperty<StoriesEntry[]>`.
The public interface would **only** have `webpackFinal` and `viteFinal` be function (because those need to take the config supplied from addons and give the user the option of modifying them.

So how to move this PR forwards?
We appreciate the effort you've put into this, and we don't expect you to make the above changes (though it you're open to that, I could support you)..

I'll close the PR @joeycozza because we don't want more users to start adopting this way of defining their list of stories in `main.ts`.

Reply:
We've been using the async function for a little while now, so this is my best guess from what I can remember of why we're using it.

We're using storybook in a monorepo, and things were/are pretty slow. At one point we were desperate, and trying anything to speed up our storybook builds/running/rebuilds etc. From what I can remember, we even tried many different ways of defining the glob strings for our stories. (like would individually listing all packages explicitly help speed things up vs 1 glob with lots of `*`). 

Anyways, long story short, from what I can remember, using an async function and utilizing the fast-glob package in there and then returning the resulting array of files from fast-glob in that function sped up startup time or build time or both. (again, sorry I don't have hard numbers or more information about it)

I believe we started using the async function back in storybook 6, possible storybook 5, I can't remember. 

Saying this in writing, I realize that the difference of fast-glob and whatever tool storybook is using is likely negligible in the grand scheme of starting up storybook. I'm totally cool if the async function should only be internally used. If that is the case, should the information about it be removed from the documentation here? https://storybook.js.org/docs/react/configure/overview#with-a-custom-implementation

Reply:
We had a chat about the documentation part in a discord meeting, I'm not sure if we reached a conclusion.
I'll ping @jonniebigodes here, and ask again:

@jonniebigodes do you think we should remove the docs references?